runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java

```diff
@@ -285,12 +316,27 @@ private void updateWarning(boolean weaponSwitch)
 			{
 				if (warnedSkills.contains(skill))
 				{
-					if (weaponSwitch)
+					// If all variables have been set, this guard will be set to false, therefore, continuing the code block pertaining to messages.
+					if (!messageTickGuarded)
 					{
-						// TODO : chat message to warn players that their weapon switch also caused an unwanted attack style change
+						// If the login message has been sent, it will run the corresponding messages if it's a weaponSwitch or just a combatStyle change.
+						if (loginMessageSent)
+						{
+							if (weaponSwitch)
+							{
+								sendChatMessage("This weapon's attack style will grant you " + getAttackColor(skill) + " XP.");
+							}
+							else
+							{
+								sendChatMessage("This attack style will grant you " + getAttackColor(skill) + " XP.");
+							}
+						}
+						else
+						{
+							sendChatMessage("You logged in wielding a weapon that will grant you " + getAttackColor(skill) + " XP.");
+						}
 					}
 					warnedSkillSelected = true;
-					break;
```

Comment:
Given the chattiness of adding multiple chat messages for a single login or weapon switch event, I would like to see messages looking more like `You logged in wielding a weapon that will grant you Attack, Strength, and Defence XP.` This would likely require ditching the loop over the attack styles and trying to find an intersection between warned and active styles instead.

Reply:
I thought about it, but.. it wouldn't be that difficult.. I have an idea in mind on how to make that.. But, I'll look into it.